velox/functions/lib/Re2Functions.cpp

```diff
@@ -378,7 +379,28 @@ class LikeConstantPattern final : public VectorFunction {
  public:
   LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
```

Comment:
In case the pattern is fixed, prefix or suffix, there is no need to compile the regular expression. Hence, it would be nice to refactor the code to ensure that regular expression is compiled only if necessary.

Reply:
Thanks for the suggestion, since RE2 doesn't support the assignment operator, a constructor is needed to initialize re_ here

Reply:
Can you use std::optional?

Reply:
Thanks for the suggestion but using std::optional also doesn't help since the assignment operator is not supported and changes would be needed in re2.h. It would also be beneficial to compile the regular expression since it is used to check if the pattern is a valid one, in checkForBadPattern(re_).

Reply:
Another option is to turn LikeConstantPattern into a template with template parameter being a Matcher object. We can then have different matchers: RE2-based, fixed, prefix and suffix. The code that instantiates the LikeConstantPattern would analyze the pattern and use the right matcher to create LikeConstantPattern.